src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs

```diff
@@ -173,22 +188,24 @@ public override void BeginFinallyBlock()
             MarkLabel(currentExBlock.TryEnd);
             currentExBlock.HandleStart = DefineLabel();
             currentExBlock.HandleEnd = finallyEndLabel;
-            _cfBuilder.AddFinallyRegion(_labelTable[currentExBlock.TryStart], _labelTable[currentExBlock.TryEnd],
-                               _labelTable[currentExBlock.HandleStart], _labelTable[currentExBlock.HandleEnd]);
+            _cfBuilder.AddFinallyRegion(GetMetaLabel(currentExBlock.TryStart), GetMetaLabel(currentExBlock.TryEnd),
+                               GetMetaLabel(currentExBlock.HandleStart), GetMetaLabel(currentExBlock.HandleEnd));
             currentExBlock.State = ExceptionState.Finally;
             MarkLabel(currentExBlock.HandleStart);
+            // Stack depth for "finally" starts at zero.
+            _currentStackDepth = 0;
         }
 
-        public override void BeginScope() => throw new NotImplementedException();
+        public override void BeginScope()
+        {
+            // TODO: No-op, will be implemented wit PDB support
```

Comment:
I though throwing might not be ideal in case user only wants to emit assembly without PDB and their code already emitting scope information, what you think?